### ISSUE TITLE
In-app glossary/explanations integration

### ISSUE THREAD

**Issue by hauf-toni** [digitalfabrik/integreat-app#2991]
**Is your feature request related to a problem? Please describe.**
**The feature should be called `explanations`.**
Users may encounter specific terms within the content of the app that they are unfamiliar with. Currently, there is no way to provide instant explanations for these terms within the app, which can hinder user understanding and accessibility. 

**Describe the solution you'd like**
Implement a frontend feature that highlights glossary terms directly within the app content. When a user encounters a term that exists in the glossary, it should be displayed in a distinct style (e.g., underlined or highlighted, as proposed by ui-ux). Tapping on the highlighted term should open a small popup or modal providing the definition of the term.

This feature would involve:

- Fetching the glossary terms from the backend.
- Automatically detecting and highlighting terms within the app content.
- Displaying a tooltip, popup, or modal with the term definition when the user taps on the highlighted word.

**Describe alternatives you've considered**
Another option would be to use a glossary icon next to the term instead of direct highlighting, which could reduce visual clutter but might be less intuitive.

**Additional context**
Backend-Issue: [#3092](https://github.com/digitalfabrik/integreat-cms/issues/3092) and [#3090](https://github.com/digitalfabrik/integreat-cms/issues/3090)

**Comment by hauf-toni** [comment]
Design-Brainstorm:
- Display glossary terms with **dotted undeline** by default.
- Display a tooltip **on hover** with the term definition.
- Open a modal **on click** with more detailed information or navigation to the glossary page.

**Comment by steffenkleinle** [comment]
>     * Display a tooltip **on hover** with the term definition.

On native/mobile this would just not be implemented, correct? Or can you think of any other solution?

**Comment by hauf-toni** [comment]
No final decisions have been made yet, I'll let you know as soon as a design is available here @steffenkleinle

**Comment by JoeyStk** [comment]
We renamed this feature on the CMS side to term explanations (=Begriffslexikon) to avoid some confusion with the glossary of MT providers (e.g. DeepL glossary). I think it could be a good idea if we use the same terminology. What do you think?  :)

**Comment by hauf-toni** [comment]
> > ```
> > * Display a tooltip **on hover** with the term definition.
> > ```
> 
> On native/mobile this would just not be implemented, correct? Or can you think of any other solution?

Yes, on mobile we would probably just skip this and show the explanation on click.

**Comment by steffenkleinle** [comment]
> We renamed this feature on the CMS side to term explanations (=Begriffslexikon) to avoid some confusion with the glossary of MT providers (e.g. DeepL glossary). I think it could be a good idea if we use the same terminology. What do you think? :)

Sounds good to me!

**Comment by steffenkleinle** [comment]
Not planned at the moment, see discussion in the CMS issue: https://github.com/digitalfabrik/integreat-cms/issues/3092
